src/main/java/Prune/Parser.java

```diff
@@ -0,0 +1,255 @@
+            }
+            Task task = this.tasks.tasksList.get(taskIndex);
+            boolean isDone;
+            isDone = command.equals(MARK_DONE_COMMAND);
+            this.tasks.markTask(task, isDone);
+            return task;
+        } catch (NumberFormatException e) {
+            throw new NumberFormatException("Task number must be an integer.");
+        }
+    }
+
+    /**
+     * Processes the delete command and removes the specified task from task list.
+     *
+     * @param input Array containing the delete command and task number.
+     * @return The task that was deleted.
+     * @throws InvalidTaskNumber If the task number is invalid.
+     */
+    public Task processDeleting(String[] input) throws InvalidTaskNumber {
+        try {
+            int taskIndex = Integer.parseInt(input[1]) - 1;
+            if (taskIndex >= this.tasks.tasksList.size() || taskIndex < 0) {
+                throw new InvalidTaskNumber("Please enter a valid task number");
+            }
+            Task task = this.tasks.tasksList.get(taskIndex);
+            this.tasks.deleteTask(task);
+            return task;
+        } catch (NumberFormatException e) {
+            throw new NumberFormatException("Task number must be an integer.");
+        }
+    }
+
+    /**
+     * Processes the adding command and creates a task based on the command type (todo, deadline, event).
+     *
+     * @param input Array containing the command and task arguments.
+     * @return Added task.
+     */
+    public Task processAdding(String[] input) {
+        String command = input[0];
+        try {
+            String details = input[1];
+            String[] inputArray;
+            String description;
+            Task task;
+            if (command.equals(TODO_COMMAND)) {
+                // Process Todo
+                task = new ToDo(details);
+            } else if (command.equals(DEADLINE_COMMAND)) {
+                // process Deadline
+                inputArray = details.split(" /by ");
+                description = inputArray[0];
+                String by = inputArray[1];
+                task = new Deadline(description, by);
+            } else {
+                // Process Event
+                inputArray = details.split(" /from ");
+                description = inputArray[0];
+                String[] fromToArray = inputArray[1].split(" /to ");
+                String from = fromToArray[0];
+                String to = fromToArray[1];
+                task = new Event(description, from, to);
+            }
+            this.tasks.addTask(task);
+            return task;
+        } catch (Exception e) {
+            System.out.println(String.format("\tInsufficient descriptions given for command: %s\n\t" + "Please ensure the following format:\n" + "\t\ttodo description\n" + "\t\tdeadline description /by when\n" + "\t\tevent description /from start /to end", command));
```

Comment:
long line alert - remember what is the limit for the number of characters in a line?